include/circt/Dialect/FIRRTL/FIRRTLAnnotationHelper.h

```diff
@@ -84,6 +84,33 @@ struct AnnoTargetCache {
     return targets.lookup(name);
   }
 
+  /// Replace `oldOp` with `newOp` in the target cache. The new and old ops can
+  /// have different names.
+  void replaceOp(Operation *oldOp, Operation *newOp) {
+    auto namOp = dyn_cast<FNamableOp>(oldOp);
+    if (!namOp)
+      return;
+    if (auto name = namOp.getNameAttr(); name && !name.getValue().empty())
+      targets.erase(name);
+    if (auto name = cast<FNamableOp>(newOp).getNameAttr();
```

Comment:
I don't know the use-case, but presumably this is for keeping the annotation target cache up to date while adding/replacing operations-- is it guaranteed the nameable operations are only replaced with other nameable operations? (re:`cast`).